plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -135,56 +147,100 @@ instance Pretty RecordSelectorExpr where
 instance NFData RecordSelectorExpr where
   rnf = rwhnf
 
+data ORDResolveData = ORDRD {
+  uri      :: Uri
+, uniqueID :: Int
+} deriving (Generic, Show)
+instance ToJSON ORDResolveData
+instance FromJSON ORDResolveData
+
 descriptor :: Recorder (WithPriority Log) -> PluginId
                 -> PluginDescriptor IdeState
 descriptor recorder plId = (defaultPluginDescriptor plId)
     { pluginHandlers =
-        mkPluginHandler SMethod_TextDocumentCodeAction codeActionProvider
+      mkPluginHandler SMethod_TextDocumentCodeAction codeActionProvider
+      <> mkPluginHandler SMethod_CodeActionResolve resolveProvider
+
     , pluginRules = collectRecSelsRule recorder
     }
 
+resolveProvider :: PluginMethodHandler IdeState 'Method_CodeActionResolve
+resolveProvider ideState pId ca@(CodeAction _ _ _ _ _ _ _ (Just resData)) =
+    pluginResponse $ do
+        case fromJSON $ resData of
+            Success (ORDRD uri int) -> do
+                nfp <- getNormalizedFilePath uri
+                CRSR _ crsDetails exts <- collectRecSelResult ideState nfp
+                pragma <- getFirstPragma pId ideState nfp
+                let pragmaEdit =
+                        if OverloadedRecordDot `elem` exts
+                        then Nothing
+                        else Just $ insertNewPragma pragma OverloadedRecordDot
+                    edits (Just crs) = convertRecordSelectors crs : maybeToList pragmaEdit
+                    edits _ = []
+                    changes = Just $ WorkspaceEdit
+                                    (Just (Map.singleton (fromNormalizedUri
+                                            (normalizedFilePathToUri nfp))
+                                            (edits (IntMap.lookup int crsDetails))))
+                                    Nothing Nothing
+                pure $ ca {_edit = changes}
+            _ -> pure ca
+
 codeActionProvider :: PluginMethodHandler IdeState 'Method_TextDocumentCodeAction
 codeActionProvider ideState pId (CodeActionParams _ _ caDocId caRange _) =
     pluginResponse $ do
         nfp <- getNormalizedFilePath (caDocId ^. L.uri)
         pragma <- getFirstPragma pId ideState nfp
-        CRSR crsMap exts <- collectRecSelResult ideState nfp
-        let pragmaEdit =
+        caps <- lift getClientCapabilities
+        CRSR crsMap crsDetails exts <- collectRecSelResult ideState nfp
+        let supportsResolve :: Maybe Bool
+            supportsResolve = caps ^? L.textDocument . _Just . L.codeAction . _Just . L.dataSupport . _Just
```

Comment:
This is checking for `dataSupport`, which we do need, but we also need `resolveSupport.properties` to contain "edit".

Reply:
Theoretically yes, however, I think currently edit is the only widely supported resolvable field, and everyone that supports something else also supports edit

Reply:
We should still do it right :) And we definitely need to check that `resolveSupport` is there. A client could have data support but not resolve support! (although that's weird)